guides/calculators/overview.md

```diff
@@ -0,0 +1,114 @@
+# Overview
+
+All calculators in Solidus inherit from the `Spree::Calculator` base class. They
+calculate and return the value of promotions, taxes, and shipping charges.
+Solidus includes built-in calculators for common types of calculations, such as
+flat rate shipping, percentage discounts, sales tax, and value-added tax (VAT).
+
+Whenever you create a new shipping method, tax rate, or promotion action, you
+also create a new instance of a `Spree::Calculator`.
+
+For example, if you create a new shipping method called "USPS Ground" that
+charges a flat rate of $10 USD, you would be creating an instance of the
+`Spree::Calculator::Shipping::FlatRate` calculator:
+
+- The new calculator instance would have `calculable_type` of
+  `Spree::ShippingMethod` (because it is associated with a shipping method, not a
+  tax rate or promotion action).
+- The new calculator instance's `preferences` attribute would have a value of
+  `{:preference=>10, :currency=>"USD"}`.
+
+## Attributes
+
+`Spree::Calculator` objects have the following attributes:
+
+- `type`: The type of calculator being used. For example, if the object
+  calculates shipping, it could be any available shipping calculator, such as
+  `Spree::Calculator::Shipping::FlatRate`.
+- `calculable_type` and `calculable_id`: The calculable type and its matching
+  ID. For example, if the object calculates shipping, the calculable would be
+  from the `Spree::ShippingMethod` model. See [calculables](#calculables) for
+  more information.
+- `preferences`: A hash of the calculator's preferences and the values of those 
+  preferences. Each type of calculator has its own preferences. See
+  [preferences](#preferences) for more information.
+
+## Calculables
+
+A calculable is an object that can be calculated (by a `Spree::Calculator`).
+The following classes are each a different `calculable_type`: 
+
+- `Spree::ShippingMethod`
+- `Spree::TaxRate`
+- `Spree::PromotionAction`
+
+A calculable `include`s the `Spree::CalculatedAdjustments` module. This module
+requires that each calculable has one calculator object. So, for each calculable
+object, an instance of a `Spree::Calculator` should also be created.
+
+For example, a shipping method called "USPS Ground" charges a flat rate of $10
+USD. The shipping method is calculable and requires an associated calculator.
+So, the rate for each shipment is calculated by the associated
+`Spree::Calculator::Shipping::FlatRate` object.
+
+Similarly, each tax rate in your store is calculable. So, instance of the
+`Spree::Calculator::DefaultTax` calculator is created and calculates the amount
+of tax that should be applied to line items, shipments, or orders.
+
+## Preferences
+
+Each `Spree::Calculator` has a `preferences` attribute. Its value is a hash that
```

Comment:
Static model preferences should be introduced here.

Reply:
Good call. Some initial docs I wrote introducing static model preferences was just merged, so I will link there.